pkg/common/telemetry/inmem.go

```diff
@@ -78,20 +77,20 @@ func (i *inmemRunner) run(ctx context.Context) error {
 	return nil
 }
 
-func (i *inmemRunner) startConfigWarning(ctx context.Context, wg *sync.WaitGroup) {
+func (i *inmemRunner) startConfigWarning(parent context.Context, wg *sync.WaitGroup) {
 	wg.Add(1)
-	sigChannel := make(chan os.Signal, 1)
-	signal.Notify(sigChannel, metrics.DefaultSignal)
+	ctx, cancel := signal.NotifyContext(parent, metrics.DefaultSignal)
```

Comment:
This change is also incorrect since it will lead to a for-select spin loop with the cancelled context. The current code is correct.

Can we revert this change?

Reply:
https://github.com/spiffe/spire/pull/2901/commits/5df363bb1503ceccdb82eeebf44210fa5284c8ac